// @cloudinary/url-gen/actions/effect.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var effect = require('../effect-de163495.cjs');
require('../Blur-96aff6a9.cjs');
require('../Qualifier-6633a22f.cjs');
require('../QualifierValue-e770d619.cjs');
require('../QualifierModel-0923d819.cjs');
require('../unsupportedError-74070138.cjs');
require('../Action-34aa7481.cjs');
require('../FlagQualifier-0e14a6c3.cjs');
require('../region-a15cac69.cjs');
require('../NamedRegion-8290d5fc.cjs');
require('../RectangleRegion-fff99f96.cjs');
require('../Accelerate-136bb4e5.cjs');
require('../Loop-7cec7bd3.cjs');
require('../LeveledEffectAction-7af017df.cjs');
require('../SimpleEffectAction-36492d76.cjs');
require('../internalConstants-4e8f184f.cjs');
require('../Cartoonify-09deff4d.cjs');
require('../Outline-5699255a.cjs');
require('../prepareColor-c03e99eb.cjs');
require('../MakeTransparent-491df0f4.cjs');
require('../Vectorize-61c48b0e.cjs');
require('../SimulateColorBlind-496c47d6.cjs');
require('../EffectActionWithLevel-1c80b4cf.cjs');
require('../AssistColorBlind-0e74e40a.cjs');
require('../GradientFade-9d24c805.cjs');
require('../FadeOut-7a32a2a3.cjs');
require('../Colorize-1b1ee6ed.cjs');
require('../Shadow-113eaac2.cjs');
require('../StyleTransfer-1bcf1cb7.cjs');
require('../Dither-adef18eb.cjs');
require('../Deshake-58641f69.cjs');
require('../Pixelate-d93012e4.cjs');
require('../EffectActionWithStrength-b3d0fc83.cjs');
require('../Blackwhite-cbd26bb2.cjs');
require('../FadeIn-6762b467.cjs');
require('../RemoveBackgroundAction-7bbffcc9.cjs');
require('../Theme-3166e9c9.cjs');
require('../BackgroundRemoval-59ada0a2.cjs');
require('../DropShadow-1aecd208.cjs');
require('../GenerativeRemove-5db6c7f9.cjs');
require('../GenerativeReplace-0f31071a.cjs');



exports.Effect = effect.Effect;
exports.accelerate = effect.accelerate;
exports.advancedRedEye = effect.advancedRedEye;
exports.artisticFilter = effect.artisticFilter;
exports.assistColorBlind = effect.assistColorBlind;
exports.backgroundRemoval = effect.backgroundRemoval;
exports.blackwhite = effect.blackwhite;
exports.blur = effect.blur;
exports.boomerang = effect.boomerang;
exports.cartoonify = effect.cartoonify;
exports.colorize = effect.colorize;
exports.deshake = effect.deshake;
exports.dither = effect.dither;
exports.dropShadow = effect.dropShadow;
exports.fadeIn = effect.fadeIn;
exports.fadeOut = effect.fadeOut;
exports.generativeRemove = effect.generativeRemove;
exports.generativeReplace = effect.generativeReplace;
exports.gradientFade = effect.gradientFade;
exports.grayscale = effect.grayscale;
exports.loop = effect.loop;
exports.makeTransparent = effect.makeTransparent;
exports.negate = effect.negate;
exports.noise = effect.noise;
exports.oilPaint = effect.oilPaint;
exports.outline = effect.outline;
exports.pixelate = effect.pixelate;
exports.redEye = effect.redEye;
exports.removeBackground = effect.removeBackground;
exports.reverse = effect.reverse;
exports.sepia = effect.sepia;
exports.shadow = effect.shadow;
exports.simulateColorBlind = effect.simulateColorBlind;
exports.styleTransfer = effect.styleTransfer;
exports.theme = effect.theme;
exports.transition = effect.transition;
exports.vectorize = effect.vectorize;
exports.vignette = effect.vignette;
